extensions/ui/org.eclipse.smarthome.ui.basic/web-src/smarthome.js

```diff
@@ -1589,7 +1591,9 @@
 			smarthome.dataModel = {};
 
 			function appendControl(control) {
-				smarthome.dataModel[control.id] = control;
+				if (control.id.length > 0) {
```

Comment:
Any reason to perform this check? Can widget ID ever be empty?

I can test it later today, as for now I only have this question.

Reply:
Yes it happens when a frame is created when not defined in the sitemap, like for example in the "Astronomical Data" or 'Group Demo' pages in the demo sitemap.